docs/code-usage.rst

```diff
@@ -0,0 +1,178 @@
+
+    $app = new \Phinx\Console\PhinxApplication($version);
+    $phinx = new \Phinx\Wrapper\TextWrapper($app);
+
+Where ``$version`` is the version number of Phinx (eg. '0.4.1').
+``$phinx`` now contains our Phinx object which will expose to us the commands.
+
+
+
+Custom Options
+---------------
+
+Phinx comes with some sensible options set for us, however these are not always appropriate and may need changing.
+
+There are two ways an option value can be set.
+
+The first way is instead of the ``$phinx`` declaration above, we can pass through a second parameter ``$options`` as an array of options.
+
+.. code-block:: php
+
+    $phinx = \Phinx\Wrapper\TextWrapper($app, array(
+        'option_name' => 'option_value',
+    ));
+
+To set multiple options and values, simply add more key value pairs to the ``$options`` array.
+
+The second way uses the original ``$phinx`` object and uses the ``setOption($option, $value)`` method.
+
+This method can be invoked multiple times, and either chained together or with each invokation done separately.
+
+.. code-block:: php
+
+    $phinx->setOption('option_name', 'option_value');
+
+The three options we can alter are:
+
+1. Environment name
+2. Configuration File Location
+3. Parser Information
+
+.. note::
+
+    There is no need to define these options if they are the same as the defaults.
+
+Configuring Environment Name
+-----------------------------
+
+Setting the Environment Name is done using the option name ``environment``.
+The value you assign should match the config file environment declared (eg. ``development``)
+
+.. code-block:: php
+
+    $phinx->setOption('environment', 'development');
+
+The default value if left blank is: ``development``.
+
+Configuring Configuration File Location
+----------------------------------------
+
+Setting the Configuration File Location is done using the option name ``configuration``.
+The value you assign should match the relative or absolute path to the phinx.yml (or alternative) file.
+
+.. code-block:: php
+
+    $phinx->setOption('configuration', '/var/www/phinx.yml');
+
+.. note::
+
+    The location may be relative to the webroot if you use an MVC framework.
+    Using ``__DIR__`` for relative paths or an absolute path is recommended.
+
+The default value if left blank is: ``./phinx.yml``.
+
+Configuring Parser Information
+-------------------------------
+
+Setting the Parser Information is done using the option name ``parser``.
+The value you assign should match the file format of the Configuration File.
+
+Possible values are:
+
+* yaml (.yml files)
+* php (.php files)
+* json (.json files)
+
+.. code-block:: php
+
+    $phinx->setOption('parser', 'yaml');
+
+.. note::
+
+    If left blank, the value is assumed based on the Configuration File's file extension as per the list above.
+
+Migrating
+----------
+
+This command will run the same command as ``phinx migrate``.
+
+.. code-block:: php
+
+    $phinx->getMigrate($env, $target);

```

Comment:
To me, what makes sense is to have 1 and 2 as you specified. However, in a way that the message is also split down and segmented into an array of messages and timestamps etc, so that if the user wished to get the response, such as 'xxx migrated 0.2s' they can simply call a method like $phinx->getMessage(); and it will have the same message as $phinx->migrate() currently has, and with work done on the message too, we can provide it in many formats / split up.

I have to agree with the use of a library rather than a command.